libfreerdp/core/transport.c

```diff
@@ -404,14 +405,25 @@ BOOL transport_connect(rdpTransport* transport, const char* hostname,
 	}
 	else
 	{
-		sockfd = freerdp_tcp_connect(context, settings, hostname, port, timeout);
+		UINT16 peerPort;
+		const char *peerHostname;
+		BOOL isProxyConnection = proxy_prepare(settings, &peerHostname, &peerPort, TRUE);
```

Comment:
Attention, the last argument should be FALSE here

Reply:
Oops.. that's a bug in my implementation of `proxy_prepare`. Probably `isHTTP` should be handled correctly there! Would you mind adding such an if-statement to the PR?

But, just for the sake of "semantically correct code", why do you pass TRUE then at all, if it changes nothing? Do you explicitly want the environment variables to be picked up? And, what is the initial comment you are referring to?

A decent url parser would be a good idea indeed. It should be centralized for both command line and environment variable. It is not implemented like this now, only because either a public symbol would be needed in libfreerdp, or the rdpSettings have to be changed to pass the unparsed URL instead of type/host/port. I guess, this would be a task for a separate issue / PR :)

Reply:
Call of proxy_prepare() is needed to set the global variable defining the proxy to ensure the isProxy will report TRUE. This is needed to avoid the hostname resolving on local but passing the hostname via the proxy to make hostname resolve on the remote end.

My idea was also to support the ALL_PROXY env.var as curl can do but for socks5 proxy only. Without the URL parser it can make a lot of unwanted effects so it is not in scope of this issue.

My usecase is very simple. As part of my work I am using lab setups mostly on virtual machines including also the Windows VMs. These "boxes" are on non-public IPs (like 192.168...) where I can ssh onto a machine in lab having also the accessible IP. ssh can act as the sock5 anonymous proxy. Anything else is out of scope of this issue.